### PR TITLE
Fix wrong scheduled transfer payment direction on PWA

### LINKED ISSUE #3230: [Bug]: Upcoming transfer transactions show incorrect payment/debit direction in one account in PWA
### Verified issue does not already exist?

- [X] I have searched and found no existing issue
- [X] I will be providing steps how to reproduce the bug (in most cases this will also mean uploading a demo budget file)

### What happened?

To reproduce, create 2 scheduled transactions that are functionally the same (e.g. Savings -> Chequing), but with Payee/Account and +/- swapped. These two transactions are doing the same thing:

image

On Desktop, both transactions show up correctly in both accounts as depositing to Chequing as well as payment from Savings:

image
image

However on PWA they show the the payment amount as-is entered in the Scheduled rule without considering direction.

image

In the above, both scheduled transactions should show as a deposit/green in Chequing and payment/red in Savings.

Test budget attached.
[2024-08-10-Test Budget.zip](https://github.com/user-attachments/files/16570038/2024-08-10-Test.Budget.zip)

### Where are you hosting Actual?

Docker

### What browsers are you seeing the problem on?

Other

### Operating System

Mobile Device

### PR DESCRIPTION
**Pull request by joel-jeremy**
Fixes #3230 

This PR introduces a new hook `useAccountPreviewTransactions` which builds on top of the new `usePreviewTransactions` hook. This new hook centralizes the logic to inverse the payees, accounts, amounts depending on where the preview transactions are being viewed from.

### CODE REVIEW

File: packages/desktop-client/src/components/accounts/Account.tsx [hunk 1]
```diff
@@ -1651,9 +1659,6 @@ class AccountInternal extends PureComponent<
         balances={balances}
         showBalances={showBalances}
         filtered={transactionsFiltered}
-        collapseTransactions={ids =>
-          this.props.splitsExpandedDispatch({ type: 'close-splits', ids })
-        }
       >
```
Comment (jfdoming) [inline packages/desktop-client/src/components/accounts/Account.tsx:1662]: With the change from a callback to a `useEffect`, I now see a very brief flash of expanded preview transactions on desktop. It's not a big deal, but any ideas on how we can avoid that?

Comment (joel-jeremy) [reply packages/desktop-client/src/components/accounts/Account.tsx:1668]: Should be fixed now :)

**Comment by jfdoming** [comment]
Thanks for the fix! It does seem to work better now, but I'm always a little wary of `useLayoutEffect` because it causes React to do two full renders. Since we don't actually need the first render at all (we're not measuring any component's size or something, just updating the data model), do you think another way can work?

File: packages/desktop-client/src/components/accounts/Account.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import React, {
   createRef,
   useMemo,
   type ReactElement,
+  useLayoutEffect,
```
[the remark this answers is not in the document]
Comment (jfdoming) [reply packages/desktop-client/src/components/accounts/Account.tsx:8]: (related to above comment)

**Comment by joel-jeremy** [comment]
This is now ready for review again :)

**Comment by matt-fidd** [comment]
I'm seeing an error when clicking on a scheduled transaction on the mobile view

image

**Review by matt-fidd — DISMISSED** [review]
LGTM!

**Comment by joel-jeremy** [comment]
> Looks like it's present in edge too, false alarm! It is a regression though

The error is due to incorrect initial loading state. Including the fix here. Let me know if we should put it in a separate PR. :)

**Review by matt-fidd — APPROVED** [review]
Great, thank you!

**Review by matt-fidd — APPROVED** [review]

**Comment by joel-jeremy** [comment]
The update-vrt seem to made the pipeline actions stuck

**Comment by youngcw** [comment]
maybe try a dummy commit

**Comment by joel-jeremy** [comment]
It's good now :)

**Review by matt-fidd — APPROVED** [review]

**Pull request merged**